src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -65,17 +72,23 @@ public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilde
 
         private static TokenValidationParameters CreateTokenValidationParameters()
         {
-            string defaultKey = Util.GetDefaultKeyValue();
-
             var result = new TokenValidationParameters();
-            if (defaultKey != null)
+            if (SecretsUtility.TryGetEncryptionKey(out byte[] key))
             {
                 // TODO: Once ScriptSettingsManager is gone, Audience and Issuer shouold be pulled from configuration.
-                result.IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(defaultKey));
+                result.IssuerSigningKey = new SymmetricSecurityKey(key);
                 result.ValidateAudience = true;
                 result.ValidateIssuer = true;
-                result.ValidAudience = string.Format(AdminJwtValidAudienceFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName));
-                result.ValidIssuer = string.Format(AdminJwtValidIssuerFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName));
+                result.ValidAudiences = new string[]
```

Comment:
tiny performance nit: if we don't expect these arrays to change over the lifetime of a host, can we cache these arrays instead of new'ing them up for each request?

Reply:
The AddJwtBearer options config delegate above makes it hard to see, but these options are only created once when the host is built, not on each request.